### ISSUE TITLE
Training on data with a fixed order

### ISSUE THREAD

**Issue by Yu-Shi** [mlfoundations/dclm#65]
Hi authors, thanks for your previous assistance in running the code. Now I can successfully run pre-training on my data. It seems that DCLM shuffles the data during tokenization and pre-training. I wonder if there is a way to support some kind of curriculum learning, i.e. to pre-train on data with a fixed order?

**Comment by GeorgiosSmyrnis** [comment]
Hi @Yu-Shi ,

While this is not something we officially support yet, this can be done by doing the following:
- When tokenizing the data, use the `--no_shuffle` flag to disable shuffling.
- To disable shuffling in the pretraining step, you need to manually make a change in `open_lm.main`, namely change https://github.com/mlfoundations/open_lm/blob/9bb92ef1689333534b7057942a20d18a46d1fa52/open_lm/main.py#L588 to `shard_shuffle_seed = None`. Please make a note of this in any submissions you make. Also, just to let you know - because of the way webdataset works (assigning a shard to each worker) the fixed order will be affected by the number of GPUs/workers per GPU, so please keep that in mind if this is something relevant to your method.

**Comment by Yu-Shi** [comment]
@GeorgiosSmyrnis Thank you, that works!
